core/src/main/java/hudson/model/Queue.java

```diff
@@ -3009,4 +3010,48 @@ public static Queue getInstance() {
     public static void init(Jenkins h) {
         h.getQueue().load();
     }
+
+    /**
+     * Schedule <tt>Queue.save()</tt> call for near future once items change. Ignore all changes until the time the save takes place.
+     */
+    @Extension
+    @Restricted(NoExternalUse.class)
+    public static final class Saver extends QueueListener implements Runnable {
```

Comment:
Is there a reason we don't want to trigger `push()` on the other queue status changes? i.e. enter/leave blocked, enter/leave buildable etc?  Presumably we're relying on the Queue `maintain()` automatically ensuring items enter the right states correctly after changes within-queue, but what are the implications of potentially failing to persist after queued items move within categories, especially if behaviors are customized by plugins?

Reply:
@olivergondza I think maintain might not complete as normal if called before there's time to establish agent connections / etc -- I think safer to simply persist on these changes -- especially since we're only writing at a certain frequency, so the performance change is pretty minor.   Also, I'm not 100% sure that running maintain will land things in a correct state (other changes may be associated with the state changes in some plugin that's interacting with the Queue).

Don't have a *concrete* example of where that might be an issue, just feels better to take the safer approach and sidestep the risk.